apps/maestro/src/server/routers/interchainToken/searchInterchainToken.ts

```diff
@@ -431,3 +463,57 @@ async function getTokenDetails(
 
   return null;
 }
+
+/**
+ * Check if a token is registered on Stellar by directly querying the Stellar contract
+ */
+export async function getStellarTokenRegistrationDetails(
+  tokenId: string,
+  ctx: Context
+): Promise<{
+  isRegistered: boolean;
+  tokenAddress: string | null;
+  tokenManagerAddress: string | null;
+}> {
+  try {
+    const chainConfig = await getStellarChainConfig(ctx);
+    const rpcUrl = STELLAR_RPC_URLS[NEXT_PUBLIC_NETWORK_ENV];
+    // Create a network-configured Stellar contract client
+    const ITSStellarContractClient = (await Client.from({
+      contractId: chainConfig.config.contracts.InterchainTokenService.address,
+      networkPassphrase: process.env.STELLAR_NETWORK_PASSPHRASE as string,
```

Comment:
don't use `process.env` directly. we have centralized them in `env.ts`, and we handle default value there.